packages/kilt-did/src/Did.utils.ts

```diff
@@ -367,3 +447,51 @@ export function writeDidfromIdentity(
     [KeyRelationship.keyAgreement]: { ...identity.boxKeyPair, type: 'x25519' },
   })
 }
+
+export async function signWithKey(
+  toSign: Uint8Array | string,
+  key: IDidKeyDetails,
+  signer: KeystoreSigner
+): Promise<{ keyId: string; alg: string; signature: Uint8Array }> {
+  const alg = getSignatureAlgForKeyType(key.type)
+  const { data: signature } = await signer.sign({
+    publicKey: Crypto.coToUInt8(key.publicKeyHex),
+    alg,
+    data: Crypto.coToUInt8(toSign),
+  })
+  return { keyId: key.id, signature, alg }
+}
+
+export async function signWithDid(
+  toSign: Uint8Array | string,
+  did: IDidDetails,
+  signer: KeystoreSigner,
+  whichKey: KeyRelationship | IDidKeyDetails['id']
+): Promise<{ keyId: string; alg: string; signature: Uint8Array }> {
+  let key: IDidKeyDetails | undefined
+  if (Object.values(KeyRelationship).includes(whichKey as KeyRelationship)) {
+    ;[key] = did.getKeys(KeyRelationship.authentication)
```

Comment:
C: I would prefer
```suggestion
    key = did.getKeys(KeyRelationship.authentication)[0]
```

Reply:
Fixed in https://github.com/KILTprotocol/sdk-js/pull/408/commits/ed4d9c3448b4bfc329e87bdd5930b88f88287e64